airflow/www/views.py

```diff
@@ -3226,6 +3226,12 @@ def process_form(self, form, is_created):
             for key in self.extra_fields
             if key in form.data and key.startswith(f"extra__{conn_type}__")
         }
+
+        # If parameters are added to the classic `Extra` field, include these values along with
+        # custom-field extras.
+        if form.data.get("extra"):
+            extra.update(json.loads(form.data.get("extra")))
```

Comment:
This would break if `extra` is something like `"null"` or `1`.

Reply:
The test cases should also these “valid in form but invalid in logic” values.

Reply:
Are you thinking this should not fail if an invalid dictionary value is added? I assumed this would fail hard and the user would need to recreate the connection again. Happy to flash an error message but still create the connection entry or catch some exceptions and log a custom failure message for users.

Reply:
Perhaps it is a better UX for the connection entry to be created but flash an error message that the invalid `Extra` value(s) are not added.  Something like this:
![image](https://user-images.githubusercontent.com/48934154/127423013-e7efe4ea-6373-49de-ac25-471022e87d83.png)

Reply:
I think this should fail hard on the user as well, but IIUC this would result in a 500 page, which doesn’t feel right. Dropping the extra value + a flash sounds much better.